### PR TITLE
Fix #28420 - Fix 'Customise homepage' title truncate issue #28420

### LINKED ISSUE #28420: 'Customise homepage' - button text bit hidden at the bottom of the button
### Steps to reproduce

1. Open Firefox for Android 
2. Create a collection by adding some bookmarks 
3. Now open a new empty tab
4. Observe the 'Customise homepage' button

### Expected behaviour

The 'Customise homepage' button text should be at the center of the button and full text should be shown. 

### Actual behaviour

Button text is not fully shown. The text of the button is bit hidden from bottom.

### Device name

Sony Xperia 5 ii

### Android version

Android 11

### Firefox release type

Firefox

### Firefox version

108.0.1

### Firefox Details

Build ID 20221215175817
User Agent Mozilla/5.0 (Android 11; Mobile; rv:108.0) Gecko/108.0 Firefox/108.0

### Attachment 

Screenshot_20230105-232828_2

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22235)

### PR DESCRIPTION
**Pull request by gitstart**
The PR fixes an accessibility issue that is when user increases font size from phone settings the `Customise homepage` button was getting truncated from bottom.

### Fix description
Removed fix button height `36 dp`and added content padding horizontal `16.dp` and vertical `12.dp`

### Issue video
https://www.loom.com/share/a464c6de26954cdda6126cda7702521b

### Demo video after fix
https://www.loom.com/share/b64a6a092b1e4184acfd6d38744b5234

### Pull Request checklist
- [ ] **Tests:** This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots:** This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility:** The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

**QA**
- [x] QA Needed
### To download an APK when reviewing a PR (after all CI tasks finished running):

1. Click on Checks at the top of the PR page.
2. Click on the firefoxci-taskcluster group on the left to expand all tasks.
3. Click on the build-debug task.
4. Click on View task in Taskcluster in the new DETAILS section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #28420

### CODE REVIEW

**Comment by MozillaNoah** [comment]
Hi, @gitstart, thanks for your contribution! I'm not 100% this is the approach we'll need for an accessibility fix here, since a height increase will not prevent _all_ button text cut-offs, pending what the user's device font size is. I'm going to reach out to our UX team to see what the best direction will be, and I'm marking this PR as `work-in-progress` and `needs-ux` in the meantime.

**Comment by gitstart** [comment]
> Hi, @gitstart, thanks for your contribution! I'm not 100% this is the approach we'll need for an accessibility fix here, since a height increase will not prevent _all_ button text cut-offs, pending what the user's device font size is. I'm going to reach out to our UX team to see what the best direction will be, and I'm marking this PR as `work-in-progress` and `needs-ux` in the meantime.

Thank you @MozillaNoah for reviewing the PR 🙏.
Sure, you can get best directions from UX team but just for more details i have tested from mdpi to 560dpi emulators and it worked fine. hopefully all buttons which uses same class as customise homepage using will not have text-cut-off issue.

**Comment by MozillaNoah** [comment]
Hi @gitstart, I heard back from our UX team today. The approach we'd like to do here instead is replace the original implementation of a fixed height (the 36.dp) with some vertical and horizontal padding and allowing the contents of the button to dynamically grow on its own. This will have the effects of ensuring the original minimum height for users with the default font size and following the specs in our design system, but it will also unrestrict the growth of text within the buttons for users with larger font sizes. 

Essentially, what we're looking for here is:

``` 
androidx.compose.material.Button(
        ...
        modifier = Modifier.fillMaxWidth(),
        ...
        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 12.dp)
    ) {
```

instead of:
``` 
androidx.compose.material.Button(
        ...
        modifier = Modifier
            .fillMaxWidth()
            .height(XX.dp),
        ...
    ) {

Where XX is any pre-determined, fixed number.
```

This difference of approach can be validated by running previews of `Button` and temporarily changing the text style from `FirefoxTheme.typography.button` to `FirefoxTheme.typography.headline5` to simulate larger device font sizes.

| Fixed height of 40 | Removing fixed height and using padding 
| -- | -- |
| image | image |

Does this change in approach make sense?

**Comment by gitstart** [comment]
Hello @MozillaNoah,
Requested changes are done as per mentioned above. 
Please review again.
Thank you again 🙏..

File: app/src/main/java/org/mozilla/fenix/compose/button/Button.kt [hunk 1]
```diff
@@ -48,7 +42,7 @@ private fun Button(
         onClick = onClick,
         modifier = Modifier
             .fillMaxWidth()
-            .height(36.dp),
+            .padding(horizontal = 16.dp, vertical = 12.dp),
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:45]: So Composables are a little tricky with padding. Sometimes just doing `Modifier.padding(..)` is enough. Other times, when using Composables in the Material library, we have to use the `contentPadding` API that those components expose. In this case, `androidx.compose.material.Button` has a parameter called `contentPadding` that we want to use here. In essence, the padding addition should look like: 

``` 
androidx.compose.material.Button(
    ...
    modifier = Modifier.fillMaxWidth(),
    ...
    contentPadding = PaddingValues(horizontal = 16.dp, vertical = 12.dp)
) {
```

If we didn't use `contentPadding`, the padding applied via the Button's modifier would instead be _outside_ the Button's drawing area. We want this padding _inside_ the Button.

File: app/src/main/java/org/mozilla/fenix/compose/button/Button.kt [hunk 2]
```diff
@@ -5,13 +5,7 @@
...
 
 import androidx.compose.foundation.background
-import androidx.compose.foundation.layout.Arrangement
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Spacer
-import androidx.compose.foundation.layout.fillMaxWidth
-import androidx.compose.foundation.layout.height
-import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.*
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/button/Button.kt:8]: Please revert these import changes, as we don't use the wildcard import, `*`.

**Review by MozillaNoah — CHANGES_REQUESTED** [review]
Thanks for the latest changes! I've left a few comments.

**Comment by gitstart** [comment]
> Thanks for the latest changes! I've left a few comments.

Thank you @MozillaNoah for reviewing PR, 
Requested changes are addressed. Please take a look..

**Review by MozillaNoah — COMMENTED** [review]
The changes look great!  

Before we land this, can you please squash the commits and rename the resulting commit and PR to follow our recommended structure from https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request (eg: For #28420: Fix ...)?

**Comment by gitstart** [comment]
> The changes look great!
> 
> Before we land this, can you please squash the commits and rename the resulting commit and PR to follow our recommended structure from https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request (eg: For #28420: Fix ...)?

Thank you for reviewing PR @MozillaNoah,
As requested, all commits are squashed.

**Comment by MozillaNoah** [comment]
> > The changes look great!
> > Before we land this, can you please squash the commits and rename the resulting commit and PR to follow our recommended structure from https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request (eg: For #28420: Fix ...)?
> 
> Thank you for reviewing PR @MozillaNoah, As requested, all commits are squashed.

The commit message isn't quit there yet. Please update it so it follows the format in https://github.com/mozilla-mobile/shared-docs/blob/main/android/CONTRIBUTING_code.md#creating-a-pull-request. Also update the PR title to match the commit message.

**Comment by gitstart** [comment]
Hello @MozillaNoah 
Pushed squashed commits as per Doc. Please check now.
Thank you

**Review by MozillaNoah — APPROVED** [review]
This looks good! I've opened https://github.com/mozilla-mobile/fenix/pull/28637 to start the landing process for this fix.  Thanks for your contribution!

**Comment by MozillaNoah** [comment]
Hi @gitstart, it looks like the lint checker is failing in the CI, but I think it'll be a quick fix. Try running the following terminal command locally and fix any of the issues it returns and then we should be good to start landing this.

> ./gradlew ktlint detekt

**Comment by gitstart** [comment]
> Hi @gitstart, it looks like the lint checker is failing in the CI, but I think it'll be a quick fix. Try running the following terminal command locally and fix any of the issues it returns and then we should be good to start landing this.
> 
> > ./gradlew ktlint detekt

Thank you for reviewing the PR @MozillaNoah,
Lint Issue is fixed. Please check now.

**Comment by MozillaNoah** [comment]
Hey @gitstart, this landed over in https://github.com/mozilla-mobile/fenix/pull/28637, so I'm closing this PR. Thanks again for the contribution!

**Pull request closed without merging**